libr/core/cmd_debug.c

```diff
@@ -1973,17 +1972,16 @@ static void r_core_cmd_bp(RCore *core, const char *input) {
 		} else {
 			addr = r_num_math (core->num, input+2);
 			if (validAddress (core, addr)) {
-				// const int bpsz = (!strcmp (core->dbg->arch, "arm"))? 4: 1;
 				bpi = r_debug_bp_add (core->dbg, addr, hwbp, NULL, 0);
 				if (bpi) {
 					free (bpi->name);
 					if (!strcmp (input + 2, "$$")) {
 						char *newname = NULL;
-						//RFlagItem *f = r_flag_get_at (core->flags, addr);
 						RFlagItem *f = r_flag_get_i2 (core->flags, addr);
+
 						if (f) {
 							if (f->offset != addr) {
-								newname = r_str_newf ("%s+%d\n", f->name, (int)(addr - f->offset));
+								newname = r_str_newf ("%s+0x%" PFMT64x "\n", f->name, addr - f->offset);
```

Comment:
addr should be always bigger than f->offset.maybe this if should be if (f->offset < addr) to ensure this condition at runtime